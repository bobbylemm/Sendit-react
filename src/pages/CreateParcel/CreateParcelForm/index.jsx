/* eslint-disable no-nested-ternary */
import React from 'react';
import { Field, reduxForm, formValueSelector } from 'redux-form';
import { connect } from 'react-redux';
import '../../../css/createParcel.scss';
import PropTypes from 'prop-types';
import Loader from 'react-loader-spinner';
import InputField from '../../../components/Input';
import Button from '../../../components/styledComponent/Button';
import BackgroundImg from '../../../components/styledComponent/BackgroundImg';

let CreateParcelForm = ({
  handleSubmit,
  submitting,
  isLoading,
  quantity,
  weight
}) => (
  <main>
    <BackgroundImg className="create-parcel-bg">
      {isLoading && (
        <div id="loader-container">
          <Loader type="Puff" color="#FFFFFF" height="100" width="100" />
        </div>
      )}
      <div className="p-3">
        <div className="user-actions-results">
          <main className="container">
            <div className="form-card">
              <form id="form" onSubmit={handleSubmit}>
                <h2>Create a parcel order</h2>
                <div className="d-inline col-9">
                  <div className="row">
                    <div className="col-3">
                      <label className="label" htmlFor="packageName">
                        Package name
                      </label>
                    </div>
                    <div className="col-9">
                      <Field
                        type="text"
                        name="packageName"
                        id="packageName"
                        placeholder="Enter package name..."
                        component={InputField}
                        required
                      />
                    </div>
                  </div>
                  {/*  */}
                  <div className="row">
                    <div className="col-3">
                      <label className="label" htmlFor="parcelLocation">
                        Pickup location
                      </label>
                    </div>
                    <div className="col-9">
                      <Field
                        type="text"
                        name="pickupLocation"
                        id="pickupLocation"
                        component={InputField}
                        placeholder="Enter pickup location..."
                        required
                      />
                    </div>
                  </div>
                  {/*  */}
                  <div className="row">
                    <div className="col-3">
                      <label className="label" htmlFor="dropOffLocation">
                        Destination
                      </label>
                    </div>
                    <div className="col-9">
                      <Field
                        type="text"
                        name="dropOfflocation"
                        id="dropOfflocation"
                        component={InputField}
                        placeholder="Enter parcel destination..."
                        required
                      />
                    </div>
                  </div>
                  {/*  */}
                  <div className="row">
                    <div className="col-3">
                      <label className="label" htmlFor="quantity">
                        Parcel quantity
                      </label>
                    </div>
                    <div className="col-9">
                      <Field
                        type="number"
                        name="quantity"
                        id="quantity"
                        component={InputField}
                        placeholder="Enter parcel quantity"
                        required
                      />
                    </div>
                  </div>
                  {/*  */}
                  <div className="row">
                    <div className="col-3">
                      <label className="label" htmlFor="weight">
                        Weight (kg)
                      </label>
                    </div>
                    <div className="col-9">
                      <Field
                        type="number"
                        name="weight"
                        id="weight"
                        component={InputField}
                        placeholder="Enter parcel weight"
                        required
                      />
                    </div>
                  </div>
                  <div className="row">
                    <div className="col-3" />
                    <div className="col-9">
                      <Button
                        type="submit"
                        id="submit"
                        disabled={submitting}
                        className="btn w-100 ht-6">
                        Create
                      </Button>
                    </div>
                  </div>
                </div>
                <div className="d-inline col-3 t-center">
                  <h2>PRICE</h2>
                  <h3 id="parcelPrice">
                    â‚¦{quantity && weight && quantity * weight * 500}
                  </h3>
                </div>
                {/*  */}
              </form>
            </div>
          </main>
          {/* The Modal */}
        </div>
      </div>
    </BackgroundImg>
  </main>
);
const selector = formValueSelector('create-parcel-form');
CreateParcelForm = connect(state => {
  const { quantity, weight } = selector(state, 'quantity', 'weight');
  return {
    quantity,
    weight
  };
})(CreateParcelForm);

CreateParcelForm.propTypes = {
  handleSubmit: PropTypes.func,
  submitting: PropTypes.bool,
  isLoading: PropTypes.bool,
  quantity: PropTypes.number,
  weight: PropTypes.number
};

export default reduxForm({
  form: 'create-parcel-form'
})(CreateParcelForm);
