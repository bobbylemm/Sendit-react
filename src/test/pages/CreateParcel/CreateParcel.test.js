import React from 'react';
import { shallow, mount } from 'enzyme';
import { reducer as formReducer } from 'redux-form';
import { createStore, combineReducers } from 'redux';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import CreateParcelForm from '../../../pages/CreateParcel/CreateParcelForm';
import Createparcel, {
  mapDispatchToProps,
  mapStateToProps,
  CreateParcel
} from '../../../pages/CreateParcel';

const mockStore = configureMockStore();

jest.mock('axios');
// const mockStore = configureMockStore();
// test('should mount the signup container', () => {
//   const handleSubmit = jest.fn();
//   const wrapper = shallow(<SignupContainer />);
//   expect(wrapper).toMatchSnapshot();
// });
// test('should mount signup form', () => {
//   const wrapper = shallow(<SignupForm />);
//   expect(wrapper).toMatchSnapshot();
// });
describe('create parcel form component', () => {
  let wrapper;
  let history;
  let wrapper4;
  let store1;
  let handleSubmit;
  let createParcel;
  const initialState = {
    auth: {
      data: {
        user: 'dadada'
      },
      isLoading: true
    }
  };
  beforeEach(() => {
    store1 = createStore(combineReducers({ form: formReducer }));
    const user = 'dodo';
    handleSubmit = jest.fn();
    createParcel = jest.fn();
    history = { push: jest.fn() };
    wrapper = mount(
      <Provider store={store1}>
        <CreateParcelForm />
      </Provider>
    );
    wrapper4 = mount(
      <Provider store={store1}>
        <CreateParcel
          user={user}
          createParcel={createParcel}
          history={history}
          handleSubmit={handleSubmit}
        />
      </Provider>
    );
  });
  it('should match the snap shot', () => {
    const tree = shallow(
      <Provider store={store1}>
        <CreateParcelForm />
      </Provider>
    );
    expect(tree.exists()).toBe(true);
    expect(tree).toMatchSnapshot();
  });
  it('should simulate change', () => {
    const input1 = wrapper.find('input').at(3);
    const input2 = wrapper.find('input').at(4);
    const price = wrapper.find('#parcelPrice');
    input1.simulate('change', { target: { value: 1 } });
    input2.simulate('change', { target: { value: 1 } });
    wrapper.setProps({ weight: 1, quantity: 1 });
    expect(price.text()).toEqual('â‚¦500');
  });
  // it('should mount signup', () => {
  //   const store = mockStore(initialState);
  //   const wrapper2 = mount(
  //     <Provider store={store}>
  //       <Createparcel />
  //     </Provider>
  //   );
  //   expect(wrapper2.props().store.getState().auth.data.user).toBe('dadada');
  // });
  // it('should call the createParcel', () => {
  //   const form = wrapper4.find('form');
  //   const input1 = wrapper4.find('input').at(0);
  //   const input2 = wrapper4.find('input').at(1);
  //   const input3 = wrapper4.find('input').at(2);
  //   input1.simulate('change', { target: { value: 'sffffffff' } });
  //   input2.simulate('change', { target: { value: 'stufff@gmail.com' } });
  //   input3.simulate('change', { target: { value: 'sffffffff' } });
  //   form.simulate('submit');
  //   expect(createParcel).toHaveBeenCalled();
  //   // expect(history.push).toHaveBeenLastCalledWith('/');
  // });
  // describe('test mapStateToProps', () => {
  //   it('test mapStateToProps', () => {
  //     expect(mapStateToProps(initialState).user).toBe('dadada');
  //   });
  //   it('test mapDispatchToProps', () => {
  //     const dispatch = jest.fn();
  //     mapDispatchToProps(dispatch).createParcel();
  //     expect(dispatch.mock.calls[0][0]).toBeDefined();
  //   });
  // });
});
