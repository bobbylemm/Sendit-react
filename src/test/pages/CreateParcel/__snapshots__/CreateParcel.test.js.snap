// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create parcel form component should match the snap shot 1`] = `ShallowWrapper {}`;
